internal/cache/metadata/stat_cache.go

```diff
@@ -203,3 +219,46 @@ func (sc *statCacheBucketView) LookUp(
 
 	return
 }
+
```

Comment:
AddNegativeEntry function will require to set f to be nil

Reply:
negative entry and size are part of next PR, trying to make smallest PR for quicker reviews

Reply:
2 line changes I guess required for rest of the two functions, I guess we can add that in this PR so all the stat cache related changes got merged in single PR.

Reply:
it wont be just 2 lines, new unit tests are also needed to test negative case

Reply:
Oh okay sorry forgot about unit tests.